include/gzip/compress.hpp

```diff
@@ -41,68 +55,29 @@ class Compressor
             throw std::runtime_error("size may use more memory than intended when decompressing");
         }
 
-        z_stream deflate_s;
-        deflate_s.zalloc = Z_NULL;
-        deflate_s.zfree = Z_NULL;
-        deflate_s.opaque = Z_NULL;
-        deflate_s.avail_in = 0;
-        deflate_s.next_in = Z_NULL;
-
-        // The windowBits parameter is the base two logarithm of the window size (the size of the history buffer).
-        // It should be in the range 8..15 for this version of the library.
-        // Larger values of this parameter result in better compression at the expense of memory usage.
-        // This range of values also changes the decoding type:
-        //  -8 to -15 for raw deflate
-        //  8 to 15 for zlib
-        // (8 to 15) + 16 for gzip
-        // (8 to 15) + 32 to automatically detect gzip/zlib header (decompression/inflate only)
-        constexpr int window_bits = 15 + 16; // gzip with windowbits of 15
-
-        constexpr int mem_level = 8;
-        // The memory requirements for deflate are (in bytes):
-        // (1 << (window_bits+2)) +  (1 << (mem_level+9))
-        // with a default value of 8 for mem_level and our window_bits of 15
-        // this is 128Kb
-
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wold-style-cast"
-        if (deflateInit2(&deflate_s, level_, Z_DEFLATED, window_bits, mem_level, Z_DEFAULT_STRATEGY) != Z_OK)
+        std::size_t max_compressed_size = libdeflate_gzip_compress_bound(compressor_, size);
```

Comment:
See libdeflate docs on this function at https://github.com/ebiggers/libdeflate/blob/9a327aae416a71471015b70f4a10f8b10cf1db31/libdeflate.h#L69-L96

Reply:
I'm not sure we need `std::size_t max_compressed_size = libdeflate_gzip_compress_bound(compressor_, size);`. Why not just use size of a input and then `resize` to actual size ? Maybe I'm missing something but is it possible for compressed output be large then input ? /cc @springmeyer

Reply:
ok, thanks for clarification.